### ISSUE TITLE
order status

### ISSUE THREAD

**Issue by ahdinosaur** [root-systems/cobuy#21]
working on https://github.com/enspiral-root-systems/cobuy/issues/3 brought back memories of building [Cobudget](http://cobudget.co).

it'd be great to have a clear set of possible "status" that an order can be in.

the so far i made a `startDate` and `endDate` to delimit the initial "intent-casting" process, which makes the following statusus:

- pending
- open
- closed

but really i think there are more. what other modes can an order be in and what should we call all these possible states?

**Comment by ahdinosaur** [comment]
some descriptions

1. while facilitator is editing order and before it becomes live
2. after facilitator clicks 'go live' and before individual intent phase completes
3. after individual intent phase completes and before group commitment is finalized
4. after group commitment is finalized and before money is paid
5. after money is paid and before invoice is sent to supplier
6. after invoice is sent to supplier and before it is acknowledged
7. after invoice is acknowledged and before shipment is received
8. after shipment is received and before shipment is accounted
9. after shipment is accounted and before shipment is divided
10. after shipment is divided and before shipment is packed
11. after shipment is packed and while packs are being picked up
12. after all packs are picked up

i probably missed some still... :smile:

**Comment by ahdinosaur** [comment]
since this issue is now relevant again, i'd appreciate any feedback, then i reckon next step here is to give identifier names to these statuses / stages / phases / modes.

**Comment by michael-smith-nz** [comment]
Nice @ahdinosaur! This is roughly how I see them looking from the users'(not admin) point of view:

Order doesn't exist yet/being drafted
- while facilitator is editing order and before it becomes live

Order open
- after facilitator clicks 'go live' and before individual intent phase completes

Order closed
- after individual intent phase completes and before group commitment is finalized

Requires Payment
- after group commitment is finalized and before money is paid
- after money is paid and before invoice is sent to supplier

Waiting on delivery
- after invoice is sent to supplier and before it is acknowledge
- after invoice is acknowledged and before shipment is received

Packing
- after shipment is received and before shipment is accounted
- after shipment is accounted and before shipment is divided
- after shipment is divided and before shipment is packed

Pick up
- after shipment is packed and while packs are being picked up

Order finished
- after all packs are picked up

**Comment by danalexilewis** [comment]
@NotThatSmith & @ahdinosaur nice write ups.

I think the stages are slightly different depending on setup parameters - similar to the summary.

For example if purchases (schools) were buying individually then the payment stage would be order sent and maybe supplier has sent out invoices. 

Note: The wording through me here, in my mind we send suppliers **orders** for goods and they send us back an **invoice for a group** or they send us back a **confirmation of order received** and **invoices to the individuals**.  Later we will be sending a supplier an invoice for the 2% tranaction fee, but we will handle this in Xero. For me invoice == statement of requirement to pay

Likewise if the order was setup that the supplier was sending the items directly to users we wouldldnt need the last two steps.

So I think when we build this we just need to imagine that the steps are configurable :)

**Comment by ahdinosaur** [comment]
coming back from https://github.com/root-systems/cobuy/pull/268, sorry i didn't notice @NotThatSmith that you named "order open" what i call the "intent phase" (or stage). let's say you wanted to close an order full-stop, maybe the supplier went bush and will be unable to fulfill any commitments. what do you call that? i'm not saying "intent phase" is the best name, but i think "order open" is ambiguous since i think you could describe most stages along the order's progress as being "open", but maybe i'm too entrenched in precise terminology to understand as a fresh user would.

**Comment by michael-smith-nz** [comment]
> that you named "order open" what I call the "intent phase" (or stage)

My above comment was wording the user would use, not programmers(that's the disconnect I've been finding at the moment).

I think we'll need real users to choose the wording for us, at this stage we're making assumptions.
I'm assuming users will get confused with the 'intent' wording because I personally got confused with it.
I'm also assuming that 'order open' means "I'm open to making an order now" to the user.  And as I type that I realise that I'm assuming that 'individual order' would make more sense to a user than 'intent'

**Comment by ahdinosaur** [comment]
> My above comment was wording the user would use, not programmers(that's the disconnect I've been finding at the moment).

yeah. if a word doesn't make sense for users, it probably isn't a good word for the code either.

however i'm noticing a tension that seems related to the two divergent perspectives on "good design".

1. the apple approach: "good design" is intuitive, any user (even a baby) should be able to understand and interact with your design without any upfront training, it hides complexity with ease
2. the [engelbart](https://en.wikipedia.org/wiki/Douglas_Engelbart) approach: "good design" is powerful, it will require upfront training, it encompasses the necessary complexity using simple paradigms, augments human intellect to go beyond what we are capable of now

in today's age, most people assume 1 to be the accepted way of "good design". i personally follow 2.

in this case, i'm okay with using words that are not immediately intuitive, as long as once understood, they clearly encapsulate the concept in a way that is simple and not too painful to explain. i personally want to strive for a user experience that is powerful, and because of this might require some upfront training to learn. in this way, i want to teach people a new lifestyle, which means learning new concepts, new tools, and new systems.
